ssh-key/src/signature.rs

```diff
@@ -63,6 +64,26 @@ where
     }
 }
 
+/// Sign the provided message bytestring using `Self` (e.g. a cryptographic key
+/// or connection to an HSM), returning a digital signature.
+pub trait AsyncSigner<S> {
+    // Using an associated type here to force the implementor to be explicit with Send/Sync
+    /// Future type which will be returned by `try_sign`
+    type SignFuture: Future<Output = signature::Result<S>>;
+    /// Attempt to sign the given message, returning a digital signature on
+    /// success, or an error if something went wrong.
+    ///
+    /// The main intended use case for signing errors is when communicating
+    /// with external signers, e.g. cloud KMS, HSMs, or other hardware tokens.
+    fn try_sign(&self, msg: &[u8]) -> Self::SignFuture;
+}
```

Comment:
Please use the upstream trait from `async-signature` (which we plan on merging into the `signature` crate soon)